### PR TITLE
Issue #3567: added ForbidAnnotationElementValueCheck

### PR DESCRIPTION
**Pull request by rnveach**
Issue #3567 

Only 2 surprises when working on this.
* `SingleLineJavadocTest` was hiding a NPE that was not in the check. Even fixing the NPE, the input file didn't produce any violations, so I just removed the test.
* `AbstractViolationReporterTest` looks like the exception moved based on the comments.

I didn't try to regress to see how any original tests worked.

### CODE REVIEW

**Comment by rnveach** [comment]
Travis is fixed.
`ImportOrderCheckTest.testVisitTokenSwitchReflection` also needed a special change since Turkish language was interfering with the text compare.

**Comment by rnveach** [comment]
@romani Wrecker has been hanged all this time. I tried stopping and restarting and it gave the `reference is not a tree` error.

**Comment by romani** [comment]
@rnveach , please do not hesitate to report problems to wercker.
I did report few seconds ago to let them investigate a problem.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheckTest.java [hunk 1]
```diff
@@ -422,15 +444,33 @@ public void testVisitTokenSwitchReflection() {
...
-        Whitebox.setInternalState(mock, "option", importOrderOptionMock);
-
-        // expecting IllegalStateException
-        mock.visitToken(astImport);
+        final Locale defaultLocale = Locale.getDefault();
+
+        try {
+            // fix for turkey locale which changes the text importOrderOption in
+            // the exception
+            Locale.setDefault(Locale.ENGLISH);
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheckTest.java]: I do not like this fix .... please put a CS suppression on this method and restore it.
I what to review this case in separate PR, and let all other changes by applied to master.

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheckTest.java]: Done.

**Pull request merged**